### PR TITLE
Drop masked bins when rebinning binned

### PR DESCRIPTION
**Pull request by SimonHeybrock**
See title.

### CODE REVIEW

File: dataset/bin.cpp [hunk 1]
```diff
@@ -264,6 +264,23 @@ auto axis_actions(const VariableConstView &var,
   return axes;
 }
 
+auto hide_masked(
```
Comment (OwenArnold) [inline dataset/bin.cpp:267]: Is `hide_masked` the right name for this? drop_masked better?

File: dataset/test/bin_test.cpp [hunk 2]
```diff
@@ -206,3 +208,15 @@ TEST_P(BinTest, group_and_bin) {
...
   EXPECT_EQ(bin(group, {edges_x}, {}), x_group);
 }
+
+TEST_P(BinTest, rebin_masked) {
+  const auto table = GetParam();
+  auto binned = bin(table, {edges_x_coarse});
+  binned.masks().set("x-mask", makeVariable<bool>(Dims{Dim::X}, Shape{2},
+                                                  Values{false, true}));
+  if (table.dims().volume() > 0) {
+    EXPECT_NE(bin(binned, {edges_x}), bin(table, {edges_x}));
```
Comment (OwenArnold) [inline dataset/test/bin_test.cpp:218]: Might be instructive for future maintenance to demonstrate the effect of dropping the mask and reapplying the comparison.

**Review by OwenArnold — APPROVED** [review]

File: dataset/bin.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply dataset/bin.cpp:267]: The point is that it does *not* make a copy of the input with dropped masks. Instead it hides the masked events by setting the bin sizes in question to zero. No copy of the events is made.

Comment (OwenArnold) [reply dataset/bin.cpp:273]: Yes, I can see that in the implementation. Leave the name as it is.

File: dataset/test/bin_test.cpp [hunk 3]
```diff
@@ -132,7 +134,7 @@ class BinTest : public ::testing::TestWithParam<DataArray> {
 
   void expect_near(const DataArrayConstView &a, const DataArrayConstView &b) {
     // "round" last digits for approximate floating point comparison
-    const auto rounding = 100.0 * units::one * buckets::sum(a);
+    const auto rounding = 400.0 * units::one * buckets::sum(a);
```
Comment (OwenArnold) [inline dataset/test/bin_test.cpp:137]: scipp also has `is_approx` https://scipp.github.io/generated/scipp.is_approx.html?highlight=is_approx#scipp.is_approx

Comment (SimonHeybrock) [reply dataset/test/bin_test.cpp:137]: Not for data arrays, and furthermore the tolerance is absolute, but relative makes more sense here. Should probably improve the tooling to support this.

**Comment by SimonHeybrock** [comment]
@OwenArnold Can you have another look, I fixed the subtle bug for rebinning with unchanged coord length. Cheers!

File: dataset/bin.cpp [hunk 4]
```diff
@@ -197,6 +198,14 @@ template <class T, class Meta> auto extract_unbinned(T &array, Meta meta) {
   return extracted;
 }
 
+/// Combine meta data from buffer and input data array and create final output
+/// data array with binned data.
+/// - Meta data that does not depend on the buffer dim is lifted to the output
+///   array.
+/// - Any meta data depending on rebinned dimensions is dropped since it becoms
```
Comment (OwenArnold) [inline dataset/bin.cpp:205]: typo becoms -> becomes

File: dataset/bin.cpp [hunk 5]
```diff
@@ -197,6 +198,14 @@ template <class T, class Meta> auto extract_unbinned(T &array, Meta meta) {
...
+///   meaningless. Note that rebinned masks have been applied before the binning
+///   step.
+/// - If rebinning, existing meta data along unchanged dimensions is preserved.
```
Comment (OwenArnold) [inline dataset/bin.cpp:208]: 👍  for explaining logic

**Review by OwenArnold — CHANGES_REQUESTED** [review]
One minor typo, otherwise approved.

File: dataset/bin.cpp (hunk 4 above)
Comment (SimonHeybrock) [reply dataset/bin.cpp:205]: I'll fix this elsewhere, so we don't have to rerun CI.

**Pull request merged**